apps/client/src/views/admin/ListUser.vue

```diff
@@ -1,57 +1,94 @@
 <script lang="ts" setup>
-import { onBeforeMount, ref, type Ref } from 'vue'
+import { onBeforeMount, ref, watch } from 'vue'
+import { type AllUsers, formatDate } from '@cpn-console/shared'
 import { useAdminUserStore } from '@/stores/admin/user.js'
-import { formatDate } from '@cpn-console/shared'
+import { useSnackbarStore } from '@/stores/snackbar.js'
 import { copyContent } from '@/utils/func.js'
-import { getRandomId } from '@gouvminint/vue-dsfr'
+import { useUserStore } from '@/stores/user.js'
 
 const adminUserStore = useAdminUserStore()
+const snackbarStore = useSnackbarStore()
 
-const allUsers: Ref<any[]> = ref([])
+const allUsers = ref<AllUsers>([])
 
 const title = 'Liste des utilisateurs'
 const headers = [
   'Identifiant',
   'Prénom',
   'Nom',
   'E-mail',
+  'Administrateur',
   'Création',
   'Modification',
 ]
-const rows: Ref<any[][]> = ref([])
 
-const tableKey = ref(getRandomId('table'))
+const setRows = () => allUsers.value.map(({ id, firstName, lastName, email, isAdmin, createdAt, updatedAt }) => ([
+  {
+    component: 'code',
+    text: id,
+    title: 'Copier l\'id',
+    class: 'fr-text-default--info text-xs cursor-pointer',
+    onClick: () => copyContent(id),
+  },
+  firstName,
+  lastName,
+  email,
+  {
+    component: 'input',
+    type: 'checkbox',
+    checked: isAdmin,
+    'data-testid': `${id}-is-admin`,
+    class: 'fr-checkbox-group--sm',
+    title: isAdmin ? `Retirer le rôle d'administrateur de ${email}` : `Donner le rôle d'administrateur à ${email}`,
+    onClick: async (event: Event & { target: { checked: boolean }}) => {
+      const value = event.target.checked
+      if (value !== isAdmin) {
+        await adminUserStore.updateUserAdminRole(id, value)
+        snackbarStore.setMessage(value ? `Le rôle d'administrateur a été attribué à ${email}` : `Le rôle d'administrateur a été retiré à ${email}`, 'success')
+        await getAllUsers()
+        // Redirect user to home if he removed himself from admin group
+        // TODO : router.push ne suffit pas, il faut un rechargement complet
+        // instance keycloak ne semble pas au courant du changement de groupe, visible au reload
+
+        // useUserStore().setUserProfile()
+        // console.log(useUserStore().userProfile?.groups)
+
+        // @ts-ignore
+        if (useUserStore().userProfile?.id === id && !value) window.location = '/'
```

Comment:
pour info @ArnaudTA @this-is-tobi si vous avez une idée
igne 57 c'est un workaround qui fonctionne

Reply:
Hmm, peut-être en faisant un `keycloak.clearToken()` @clairenollet ?

Reply:
On s'en occupera plus tard si ton workaround fonctionne, on pourrait le faire en même temps qu'un tour des remontées Sonar par exemple